core/src/transaction_ffi.rs

```diff
@@ -0,0 +1,324 @@
+//! Module to provide easy construction of C-compatible interfaces for
+//! interacting with transactions from valid transaction references.
+//!
+
+use {
+    agave_transaction_ffi::{
+        AccountCallback, Instruction, InstructionCallback, TransactionInterface, TransactionPtr,
+    },
+    solana_svm_transaction::svm_transaction::SVMTransaction,
+};
+
+/// Given a reference to any type that implements `SVMTransaction`, create a
+/// `TransactionInterface` that can be used to interact with the transaction in
+/// C-compatible code. This interface is only valid for the lifetime of the
+/// transaction reference, which cannot be guaranteed by this function interface.
```

Comment:
not sure this is strictly true, possibly could add an unsized marker to the `TransactionInterface` to hold the lifetime 🤔